cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/gvn/internal/ValueNumberingInternal.qll

```diff
@@ -41,8 +41,75 @@ newtype TValueNumber =
   ) {
     loadTotalOverlapValueNumber(_, irFunc, type, memOperand, operand)
   } or
+  TCallValueNumber(TCallPartialValueNumber vn) { callValueNumber(_, _, vn) } or
   TUniqueValueNumber(IRFunction irFunc, Instruction instr) { uniqueValueNumber(instr, irFunc) }
 
+newtype TCallPartialValueNumber =
+  TNilArgument() or
+  TArgument(TCallPartialValueNumber head, TValueNumber arg) {
+    exists(CallInstruction call, int index |
+      callArgValueNumber(call, index, arg) and
+      callPartialValueNumber(call, index, head)
+    )
+  }
+
+predicate callValueNumber(CallInstruction call, int index, TCallPartialValueNumber vn) {
```

Comment:
Should we maybe exclude `CallInstruction`s that has a void return type? That's a slightly less drastic change, and it'll suffice for our vector size use-case.

Reply:
Also, I think all of the new definitions can be marked as `private`, no?

Reply:
Actually the place to do this is in CallPartialValueNumber